wrapper/src/main/java/com/moodysalem/phantomjs/wrapper/beans/RenderOptions.java

```diff
@@ -1,61 +1,92 @@
 package com.moodysalem.phantomjs.wrapper.beans;
 
 import com.moodysalem.phantomjs.wrapper.enums.RenderFormat;
+import java.util.Set;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import org.hibernate.validator.internal.engine.ConstraintViolationImpl;
 
 public class RenderOptions {
-    public static final RenderOptions EMPTY = new RenderOptions(null, null, null, null, null, null, null, null, null);
-    public static final RenderOptions DEFAULT =
-        EMPTY.withMargin(Margin.ZERO)
-            .withDimensions(ViewportDimensions.VIEW_1280_1024)
-            .withPaperSize(PaperSize.Letter)
-            .withRenderFormat(RenderFormat.PDF)
-            .withFooterInfo(BannerInfo.EMPTY)
-            .withHeaderInfo(BannerInfo.EMPTY)
-            .withJavaScriptExecutionDetails(1000L, 100L)
-            .withPhantomJSOptions(PhantomJSOptions.DEFAULT.withDiskCache(true));
-
-    private final PhantomJSOptions options;
-    private final PaperSize paperSize;
-    private final ViewportDimensions dimensions;
-    private final Margin margin;
-    private final BannerInfo headerInfo;
-    private final BannerInfo footerInfo;
-    private final RenderFormat renderFormat;
-    private final Long jsWait;
-    private final Long jsInterval;
-
-    private RenderOptions(PhantomJSOptions options, PaperSize paperSize, ViewportDimensions dimensions, Margin margin, BannerInfo headerInfo, BannerInfo footerInfo, RenderFormat renderFormat, Long jsWait, Long jsInterval) {
-        this.options = options;
-        this.paperSize = paperSize;
-        this.dimensions = dimensions;
-        this.margin = margin;
-        this.headerInfo = headerInfo;
-        this.footerInfo = footerInfo;
-        this.renderFormat = renderFormat;
-        this.jsWait = jsWait;
-        this.jsInterval = jsInterval;
+
+    private static Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
```

Comment:
I think either mutable or immutable is fine. I didn't see any benefit in mutable object having fluent API, so I changed it to a fitting mutable one. And I didn't like the awfully long parameter list in the constructor that was the reason I chose a mutable object over an immutable.